content/guides/reader_conditionals.adoc

```diff
@@ -0,0 +1,198 @@
+
+[source,clojure]
+----
+(deftest reader-conditionals
+     ;; snip
+     (testing "splicing"
+              (is (= [] [#?@(:clj [])]))
+              (is (= [:a] [#?@(:clj [:a])]))
+              (is (= [:a :b] [#?@(:clj [:a :b])]))
+              (is (= [:a :b :c] [#?@(:clj [:a :b :c])]))
+              (is (= [:a :b :c] [#?@(:clj [:a :b :c])]))))
+----
+
+== File organisation
+
+There isn't a clear community consensus yet around where to put `.cljc` files. Two options are to have a `src` directory where `.clj`, `.cljs`, and `.cljc` files, or to have separate `src/clj`, `src/cljc`, and `src/cljs` directories.
+
+== CLJX
+
+Before Reader Conditionals, the same goal of sharing code between platforms was solved by https://github.com/lynaghk/cljx[cljx]. cljx processes Clojure files with a .cljx extension and output multiple platform specific files to a generated sources directory. These were then read as normal Clojure or ClojureScript files by the Clojure <<xref/../../reference/reader#,reader>>. This worked well, but required another piece of tooling to run, and it wasn't able to be used in Clojure contrib projects. cljx was deprecated on June 13 2015 in favour of reader conditionals.
+
+Sente uses CLJX for sharing code between Clojure and ClojureScript. I've rewritten the https://github.com/ptaoussanis/sente/blob/v1.4.1/src/taoensso/sente.cljx[main] namespace to use reader conditionals. Notice that we've used the splicing reader conditional to splice the vector into the parent `:require`. Notice also that some of the requires are duplicated between `:clj` and `:cljs`.
+
+[source,clojure]
+----
+(ns taoensso.sente
+  (:require
+    #?@(:clj  [[clojure.string :as str]
+               [clojure.core.async :as async]
+               [taoensso.encore :as enc]
+               [taoensso.timbre :as timbre]
+               [taoensso.sente.interfaces :as interfaces]]
+        :cljs [[clojure.string :as str]
+               [cljs.core.async :as async]
+               [taoensso.encore :as enc]
+               [taoensso.sente.interfaces :as interfaces]]))
+  #?(:cljs (:require-macros
+             [cljs.core.async.macros :as asyncm :refer (go go-loop)]
+             [taoensso.encore :as enc :refer (have? have have-in)])))
+----
+
+
+== Backwards compatibility
+
+At the time of writing, there is no way to use `.cljc` files in versions of Clojure less than 1.7, nor is there any porting mechanism to preprocess `.cljc` files like CLJX does. For that reason library maintainers may need to wait for a while until they can safely drop support for older versions of Clojure and adopt reader conditionals.
```

Comment:
"nor is there any porting mechanism to preprocess `.cljc` files like CLJX does" -- not sure I understand. `cljx` still works with Clojure 1.7, doesn't it? Also "cljx" or "CLJX" are both used in this guide -- you should pick one (the former?).